timesketch/api/v1/resources/sigma.py

```diff
@@ -87,15 +125,274 @@ def get(self, rule_uuid):
+        """
+
+        rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+
+        if not rule:
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                "No rule found with rule_uuid.{0!s}".format(rule_uuid),
+            )
+
+        db_session.delete(rule)
+        db_session.commit()
+
+        return HTTP_STATUS_CODE_OK
+
+    @login_required
+    def post(self, rule_uuid=None):
+        """Handles POST request to the resource.
+        Handels POST API calls to /sigmarule/<string:rule_uuid>/ where the
+        rule_uuid is the primaray way to identify the rule.
+        The remaining attributes of the rule are provided in request itself.
+
+        If no `rule_yaml` is found in the reuqest, the method will fail as this
+        is required to parse the rule.
+
+        Args:
+            rule_uuid: uuid of the rule
+
+        Returns:
+            Sigma rule object and HTTP status code indicating
+            whether operation was sucessful.
+        """
+        form = request.json
+        if not form:
+            form = request.data
+
+        rule_yaml = form.get("rule_yaml")
+
+        if not rule_yaml:
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                "No rule_yaml supplied in the POST request to the API.",
+            )
+
+        if not isinstance(rule_yaml, str):
+            abort(
+                HTTP_STATUS_CODE_FORBIDDEN, "rule_yaml needs to be a string."
+            )
+
+        parsed_rule = ts_sigma_lib.parse_sigma_rule_by_text(rule_yaml)
+
+        if not rule_uuid:
+            rule_uuid = parsed_rule.get("id")
+
+        # Query rules to see if it already exist and exit if found
+        sigma_rule_from_db = SigmaRule.query.filter_by(
+            rule_uuid=rule_uuid
+        ).first()
+        if sigma_rule_from_db:
+            if sigma_rule_from_db.rule_uuid == parsed_rule.get("rule_uuid"):
+                logger.debug("Rule was already found in teh database")
+                # it will overwritten with the provided content
+
+        title = parsed_rule.get("title")
+        if not title:
+            logger.error("Sigma rule without title was given")
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Missing value: 'title' from the request.",
+            )
+
+        description = parsed_rule.get("description")
+        if not description:
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Missing value: 'description' from the request.",
+            )
+
+        sigma_rule = SigmaRule.get_or_create(
+            rule_uuid=rule_uuid,
+            rule_yaml=form.get("rule_yaml"),
+            description=description,
+            title=title,
+            user=current_user,
+        )
+
+        if not sigma_rule:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sigma rule object created")
+
+        sigma_rule.query_string = parsed_rule.get("es_query")
```

Comment:
Can't we add `es_query` as part of the `get_or_create` call before? Same questio for `status=experimental` you're doing in L378; this would avoid yet another query to the database.